Updated ActionColumn.php

Shifrin opened the pull request:
Enhanced for simple feature such as add HTML options to a specific
targeted button instead of overriding `initDefaultButtons()` function.

I think it will useful if not please correct me.

Shifrin committed:
Updated ActionColumn.php

Enhanced for simple feature such as add html options to a specific
targeted button instead of overriding 'initDefaultButtons ()' function.

SamMousa commented on framework/grid/ActionColumn.php:
```diff
@@ -139,7 +150,7 @@ protected function initDefaultButtons()
                     'aria-label' => Yii::t('yii', 'View'),
                     'data-pjax' => '0',
-                ], $this->buttonOptions);
+                ], isset($this->buttonOptions['view']) ? $this->buttonOptions['view'] : $this->buttonOptions);
```
This will cause weird stuff to happen if I set some button options but not others.
````
$buttonOptions['weird1'] = ['someOption' => 'someValue']
$buttonOptions['weird2'] = ['someOption' => 'someValue']
````

Will cause the following configuration for the view (and any other default) button:
````
[
    'title' => Yii::t('yii', 'View'),
    'aria-label' => Yii::t('yii', 'View'),
    'data-pjax' => '0',
    'weird1' => [someOption' => 'someValue']
    'weird2' => [['someOption' => 'someValue']
]
````

Shifrin replied:
@SamMousa Thanks for the note.

I realized it, what do you think if I removed default options?

```
$options = [
     'title' => Yii::t('yii', 'View'),
     'class' => 'btn btn-info btn-xs btn-view',
     'aria-label' => Yii::t('yii', 'View'),
     'data-pjax' => '0',
];
                
if (isset($this->buttonOptions['view'])) {
     $options = array_merge($options, $this->buttonOptions['view']);
}
```

SamMousa replied:
--disclaimer: I'm not in the Yii team --

I'd suggest introducing a different variable for default options `$defaultButtonOptions` and always merging those into whatever button configuration is passed via `$buttonOptions`. That way it's clear what the difference is.

Shifrin replied:
Thanks for the suggestion, it makes sense to me.

Shifrin committed:
Updated ActionColumn.php

Enhanced for simple feature such as add html options to a specific
targeted button instead of overriding 'initDefaultButtons ()' function.

Shifrin committed:
Revert "Updated ActionColumn.php"

Enhanced for simple feature such as add html options to a specific
targeted button instead of overriding 'initDefaultButtons()' function.

Shifrin commented:
Thanks for the reply, I'm not able to provide a right use case. Because I think the buttons property is fulfilled my needs instead of introducing a new property. Therefore I'm closing this pull request.
